ops/pebble.py

```diff
@@ -788,6 +799,13 @@ def to_yaml(self) -> str:
 
     __str__ = to_yaml
 
+    def __eq__(self, __value: Union['PlanDict', 'Plan']) -> bool:
+        if isinstance(__value, dict):
+            return self.to_dict() == __value
+        elif isinstance(__value, Plan):
+            return self.to_dict() == __value.to_dict()
+        return super().__eq__(__value)
```

Comment:
I guess not. `super` felt more correct to me ("I don't know, ask the other classes I'm made of" rather than "I don't implement this check") but `NotImplemented` is more consistent with `Layer` and is probably more expected with `__eq__` I guess. I'll change it.